gravitee-apim-console-webui/src/management/api/endpoints-v4/backend-services/endpoint-group/api-endpoint-group.component.spec.ts

```diff
@@ -0,0 +1,229 @@
+      },
+    });
+
+    fixture = TestBed.createComponent(ApiEndpointGroupComponent);
+
+    httpTestingController = TestBed.inject(HttpTestingController);
+
+    componentHarness = await TestbedHarnessEnvironment.harnessForFixture(fixture, ApiEndpointGroupHarness);
+
+    expectApiGetRequest(api);
+
+    fixture.detectChanges();
+  };
+
+  beforeEach(async () => {
+    await initComponent();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    httpTestingController.verify();
+  });
+
+  describe('GIVEN the current page is the endpoint group page', () => {
+    let api: ApiV4;
+
+    beforeEach(async () => {
+      await TestBed.compileComponents();
+      fixture = TestBed.createComponent(ApiEndpointGroupComponent);
+      httpTestingController = TestBed.inject(HttpTestingController);
+
+      api = fakeApiV4({
+        id: API_ID,
+        endpointGroups: [
+          {
+            name: DEFAULT_GROUP_NAME,
+            type: 'kafka',
+            endpoints: [],
+          },
+        ],
+      });
+      fixture.detectChanges();
+    });
+    describe('WHEN the general tab is selected', () => {
+      beforeEach(async () => {
+        await componentHarness.clickGeneralTab();
+
+        expectApiGetRequest(api);
+
+        fixture.detectChanges();
+      });
+      it('THEN the endpoint group name field should be visible with a default value', async () => {
+        expect(await componentHarness.readEndpointGroupNameInput()).toEqual(DEFAULT_GROUP_NAME);
+      });
+
+      describe('AND WHEN the endpoint group name is modified with a valid value', () => {
+        beforeEach(async () => {
+          await componentHarness.writeToEndpointGroupNameInput(VALID_ENDPOINT_GROUP_NAME);
+
+          fixture.detectChanges();
+        });
+        it('THEN the endpoint group name field should be updated with the expected value', async () => {
+          expect(await componentHarness.readEndpointGroupNameInput()).toEqual(VALID_ENDPOINT_GROUP_NAME);
+        });
+
+        it('THEN the save button should be valid', async () => {
+          expect(await componentHarness.isGeneralTabSaveButtonInvalid()).toBeFalsy();
+        });
+
+        describe('AND WHEN the save button is clicked', () => {
+          beforeEach(async () => {
+            api = fakeApiV4({
+              id: API_ID,
+              endpointGroups: [
+                {
+                  name: DEFAULT_GROUP_NAME,
+                  type: 'kafka',
+                  endpoints: [],
+                },
+              ],
+            });
+
+            await componentHarness.clickEndpointGroupSaveButton();
+
+            expectApiGetRequest(api);
+
+            fixture.detectChanges();
+          });
+          it('THEN the endpoint group name field should be saved', async () => {
+            const req = httpTestingController.expectOne({ url: `${CONSTANTS_TESTING.env.v2BaseURL}/apis/${api.id}`, method: 'PUT' });
+
+            expect(req.request.body.endpointGroups).toEqual([
```

Comment:
This test data can be extracted out into another file

Reply:
Instead of hardcoding the value, you can use the input value and change only the name with 
```
const expectedEndpointGroups: EndpointGroupV4[] = [{ ...api.endpointGroups[0], name: VALID_ENDPOINT_GROUP_NAME }];

```